### PR TITLE
test(AwsRedisInstance): refactor elastiCacheClientFake to use single mutex

### PR DESCRIPTION
**Pull request by dushanpantic**
**Description**

Changes proposed in this pull request:

- multiple mutexes were unnecessary, so they are replaced by a single mutex to improve readability

**Related issue(s)**

### CODE REVIEW

File: pkg/kcp/provider/aws/mock/elastiCacheClientFake.go [hunk 1]
```diff
@@ -45,38 +45,28 @@ func getDefaultParams() map[string]elasticacheTypes.Parameter {
...
-		securityGroupsMutex: &sync.Mutex{},
-		replicationGroups:   map[string]*elasticacheTypes.ReplicationGroup{},
-		cacheClusters:       map[string]*elasticacheTypes.CacheCluster{},
-		subnetGroups:        map[string]*elasticacheTypes.CacheSubnetGroup{},
-		parameterGroups:     map[string]*elasticacheTypes.CacheParameterGroup{},
-		parameters:          map[string]map[string]elasticacheTypes.Parameter{},
-		secretStore:         map[string]*secretsmanager.GetSecretValueOutput{},
-		userGroups:          map[string]*elasticacheTypes.UserGroup{},
-		securityGroups:      []*ec2Types.SecurityGroup{},
+		mutex:             &sync.Mutex{},
```
Comment (vandjelk) [inline pkg/kcp/provider/aws/mock/elastiCacheClientFake.go:61]: why not `m sync.Mutex`?

Comment (dushanpantic) [reply pkg/kcp/provider/aws/mock/elastiCacheClientFake.go:61]: I dont like non-descriptive names

Comment (vandjelk) [reply pkg/kcp/provider/aws/mock/elastiCacheClientFake.go:61]: I was actually pointing to both, name and the type. I was curious why pointer. I looked at the https://pkg.go.dev/sync#Mutex and `func (m *Mutex) Lock()` operates over pointer but I couldn't find reason why we should use `*sync.Mutex` and not `sync.Mutex` except that other implantations, [among go tour](https://go.dev/tour/concurrency/9), are using `sync.Mutex`. Anyway, not a big deal except that we should try to be consistent.

Comment (dushanpantic) [reply pkg/kcp/provider/aws/mock/elastiCacheClientFake.go:61]: I will change it to value

**Review by vandjelk — APPROVED** [review]

**Pull request merged**